src/sn.c

```diff
@@ -92,100 +92,137 @@ static int load_allowed_sn_community(n2n_sn_t *sss, char *path) {
+
+		case 't': /* mgmt-port */
+			sss->mport = atoi(_optarg);
+			break;
+
+		case 'd': {
+			dec_ip_str_t ip_str = {'\0'};
+			in_addr_t net;
+			uint8_t bitlen;
+
+			if (sscanf(_optarg, "%15[^/]/%hhu", ip_str, &bitlen) != 2) {
+				traceEvent(TRACE_WARNING, "Bad net/bit format '%s'. See -h.", _optarg);
+				break;
+			}
+
+			net = inet_addr(ip_str);
+			if ((net < 0) || (net == INADDR_NONE) || (net == INADDR_ANY)) {
+				traceEvent(TRACE_WARNING, "Bad network '%s' in '%s', Use default: '%s/%d'",
+				           ip_str, _optarg,
+				           N2N_SN_DHCP_NET_ADDR_DEFAULT, N2N_SN_DHCP_NET_BIT_DEFAULT);
+				break;
+			}
+
+			if (bitlen > 32) {
+				traceEvent(TRACE_WARNING, "Bad prefix '%hhu' in '%s', Use default: '%s/%d'",
+				           bitlen, _optarg,
+				           N2N_SN_DHCP_NET_ADDR_DEFAULT, N2N_SN_DHCP_NET_BIT_DEFAULT);
+				break;
+			}
+
+			traceEvent(TRACE_NORMAL, "The subnet of DHCP service is: '%s/%hhu'.", ip_str, bitlen);
+
+			sss->dhcp_addr.net_addr = ntohl(net);
+			sss->dhcp_addr.net_bitlen = bitlen;
+
+			break;
+		}
 
 #ifndef WIN32
-  case 'u': /* unprivileged uid */
-    sss->userid = atoi(_optarg);
-    break;
+		case 'u': /* unprivileged uid */
+			sss->userid = atoi(_optarg);
+			break;
 
-  case 'g': /* unprivileged uid */
-    sss->groupid = atoi(_optarg);
-    break;
+		case 'g': /* unprivileged uid */
+			sss->groupid = atoi(_optarg);
+			break;
 #endif
 
-  case 'c': /* community file */
-    load_allowed_sn_community(sss, _optarg);
-    break;
+		case 'c': /* community file */
+			load_allowed_sn_community(sss, _optarg);
+			break;
 
-  case 'f': /* foreground */
-    sss->daemon = 0;
-    break;
+		case 'f': /* foreground */
+			sss->daemon = 0;
+			break;
 
-  case 'h': /* help */
-    help();
-    break;
+		case 'h': /* help */
+			help();
+			break;
 
-  case 'v': /* verbose */
-    setTraceLevel(getTraceLevel() + 1);
-    break;
+		case 'v': /* verbose */
+			setTraceLevel(getTraceLevel() + 1);
+			break;
 
-  default:
-    traceEvent(TRACE_WARNING, "Unknown option -%c: Ignored.", (char)optkey);
-    return(-1);
-  }
+		default:
+			traceEvent(TRACE_WARNING, "Unknown option -%c: Ignored.", (char) optkey);
+			return (-1);
+	}
 
-  return(0);
+	return (0);
 }
 
+
 /* *********************************************** */
 
 static const struct option long_options[] = {
-  { "communities",     required_argument, NULL, 'c' },
-  { "foreground",      no_argument,       NULL, 'f' },
-  { "local-port",      required_argument, NULL, 'l' },
-  { "mgmt-port",       required_argument, NULL, 't' },
-  { "help"   ,         no_argument,       NULL, 'h' },
-  { "verbose",         no_argument,       NULL, 'v' },
-  { NULL,              0,                 NULL,  0  }
+		{"communities", required_argument, NULL, 'c'},
+		{"foreground",  no_argument,       NULL, 'f'},
+		{"local-port",  required_argument, NULL, 'l'},
+		{"mgmt-port",   required_argument, NULL, 't'},
+		{"dhcp",        required_argument, NULL, 'd'},
```

Comment:
Does it really need  a long option, especially as DHCP might not be the most fitting term here. Auto-IP?